src/Collection/Collection.router.ts

```diff
@@ -70,51 +96,58 @@ export class CollectionRouter extends Router {
   }
 
   async upsertCollection(req: AuthRequest) {
-    const id = server.extractFromReq(req, 'id')
-    const collectionJSON: any = server.extractFromReq(req, 'collection')
-    const eth_address = req.auth.ethAddress
-
-    const validator = ajv.compile(collectionSchema)
-    validator(collectionJSON)
-
-    if (validator.errors) {
-      throw new HTTPError('Invalid schema', validator.errors)
-    }
-
-    const canUpsert = await new Ownable(Collection).canUpsert(id, eth_address)
-    if (!canUpsert) {
-      throw new HTTPError(
-        'Unauthorized user',
-        { id, eth_address },
-        STATUS_CODES.unauthorized
+    try {
+      const id = server.extractFromReq(req, 'id')
+      const collectionJSON: any = server.extractFromReq(req, 'collection')
+      const eth_address = req.auth.ethAddress
+
+      const validator = ajv.compile(collectionSchema)
+      validator(collectionJSON)
+
+      if (validator.errors) {
+        throw new HTTPError('Invalid schema', validator.errors)
+      }
+
+      const canUpsert = await new Ownable(Collection).canUpsert(id, eth_address)
+      if (!canUpsert) {
+        throw new HTTPError(
+          'Unauthorized user',
+          { id, eth_address },
+          STATUS_CODES.unauthorized
+        )
+      }
+
+      const attributes = {
+        ...collectionJSON,
+        eth_address
+      } as CollectionAttributes
+
+      if (id !== attributes.id) {
+        throw new HTTPError('The body and URL collection ids do not match', {
+          urlId: id,
+          bodyId: attributes.id
+        })
+      }
+
+      const factoryCollection = new FactoryCollection()
+      attributes.salt = factoryCollection.getSalt(id)
+      attributes.contract_address = factoryCollection.getContractAddress(
+        attributes.salt!,
+        attributes.eth_address
       )
-    }
 
-    const attributes = {
-      ...collectionJSON,
-      eth_address
-    } as CollectionAttributes
-
-    if (id !== attributes.id) {
-      throw new HTTPError('The body and URL collection ids do not match', {
-        urlId: id,
-        bodyId: attributes.id
-      })
+      await new Collection(attributes).upsert()
```

Comment:
shouldn't we `return` here instead of `await`? I don't think we do anything with the response on the frontend anyway, but this would change the content of the response compared to the what we have in master

Reply:
You're right!